Error: The terraform-provider-azurerm_v2.96.0_x5 plugin crashed!

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* All resources and data resources

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Resource Group Creation - Default is "false"
data "azurerm_resource_group" "rgrp" {
  name = local.rg_name
}


# Get VNET 
data "azurerm_virtual_network" "vnet" {
  name                = local.vnet_name
  resource_group_name = local.rg_name
}

# Ddos protection plan - Default is "true"
resource "azurerm_network_ddos_protection_plan" "ddos" {
  count               = var.create_ddos_plan ? 1 : 0
  name                = lower("${local.vnet_name}-ddos-protection-plan")
  resource_group_name = local.rg_name
  location            = data.azurerm_virtual_network.vnet.location
  #  tags                = merge({ "ResourceName" = lower("${local.vnet_name}-ddos-protection-plan") }, var.tags, {"Environment" = upper("${var.environment}")} )
}

# Network Watcher - Default is "false"
resource "azurerm_resource_group" "nwatcher" {
  // count    = try(var.create_network_watcher, true) != false ? 1 : 0
  count    = 0
  name     = "NetworkWatcherRG"
  location = data.azurerm_virtual_network.vnet.location
  tags     = merge({ "ResourceName" = "NetworkWatcherRG" }, var.tags, )
}

resource "azurerm_network_watcher" "nwatcher" {
  // count               = try(var.create_network_watcher, true) != false ? 1 : 0
  count               = 0
  name                = "NetworkWatcher_${data.azurerm_virtual_network.vnet.location}"
  location            = data.azurerm_virtual_network.vnet.location
  resource_group_name = azurerm_resource_group.nwatcher.0.name
  tags                = merge({ "ResourceName" = format("%s", "NetworkWatcher_${data.azurerm_virtual_network.vnet.location}") }, var.tags, )
}

// # Subnets
resource "azurerm_subnet" "fw-snet" {
  name                 = "AzureFirewallSubnet"
  resource_group_name  = local.rg_name
  virtual_network_name = split("/", data.azurerm_virtual_network.vnet.id)[8]
  address_prefixes     = try(lookup(tomap(data.azurerm_subnet.subnets), "AzureFirewallSubnet").address_prefixes, [cidrsubnet(element(data.azurerm_virtual_network.vnet.address_space, 0), 3, 0)])
  service_endpoints    = try(var.firewall_service_endpoints, null)
}

resource "azurerm_subnet" "gw-snet" {
  name                 = "GatewaySubnet"
  resource_group_name  = local.rg_name
  virtual_network_name = split("/", data.azurerm_virtual_network.vnet.id)[8]
  address_prefixes     = try(lookup(tomap(data.azurerm_subnet.subnets), "GatewaySubnet").address_prefixes, [cidrsubnet(element(data.azurerm_virtual_network.vnet.address_space, 0), 4, 2)])
  service_endpoints    = try(var.gateway_service_endpoints, null)
}

data "azurerm_subnet" "subnets" {
  for_each             = local.vnets_data
  name                 = each.key
  virtual_network_name = local.vnet_name
  resource_group_name  = local.rg_name
}

resource "azurerm_subnet" "snet" {
  for_each             = try(var.subnets, {})
  name                 = lower(format("snet-%s-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}", each.value.subnet_name))
  resource_group_name  = local.rg_name
  virtual_network_name = data.azurerm_virtual_network.vnet.name
  address_prefixes     = each.value.subnet_address_prefix
  service_endpoints    = try(each.value, "service_endpoints", [])

  enforce_private_link_endpoint_network_policies = lookup(each.value, "enforce_private_link_endpoint_network_policies", null)
  enforce_private_link_service_network_policies  = lookup(each.value, "enforce_private_link_service_network_policies", null)

  dynamic "delegation" {
    for_each = try(each.value, "delegation", {}) != {} ? [1] : []
    content {
      name = lookup(each.value.delegation, "name", null)
      service_delegation {
        name    = lookup(each.value.delegation.service_delegation, "name", null)
        actions = lookup(each.value.delegation.service_delegation, "actions", null)
      }
    }
  }
}

# NSG for each subnet
resource "azurerm_network_security_group" "nsg" {
  for_each            = try(var.subnets, {})
  name                = lower("nsg_${each.key}_in")
  resource_group_name = local.rg_name
  location            = data.azurerm_virtual_network.vnet.location
  tags                = merge({ "ResourceName" = lower("nsg_${each.key}_in") }, var.tags, )
  dynamic "security_rule" {
    for_each = concat(lookup(each.value, "nsg_inbound_rules", []), lookup(each.value, "nsg_outbound_rules", []))
    content {
      name                       = security_rule.value[0] == "" ? "Default_Rule" : security_rule.value[0]
      priority                   = security_rule.value[1]
      direction                  = security_rule.value[2] == "" ? "Inbound" : security_rule.value[2]
      access                     = security_rule.value[3] == "" ? "Allow" : security_rule.value[3]
      protocol                   = security_rule.value[4] == "" ? "Tcp" : security_rule.value[4]
      source_port_range          = "*"
      destination_port_range     = security_rule.value[5] == "" ? "*" : security_rule.value[5]
      source_address_prefix      = security_rule.value[6] == "" ? element(each.value.subnet_address_prefix, 0) : security_rule.value[6]
      destination_address_prefix = security_rule.value[7] == "" ? element(each.value.subnet_address_prefix, 0) : security_rule.value[7]
      description                = "${security_rule.value[2]}_Port_${security_rule.value[5]}"
    }
  }
}

resource "azurerm_subnet_network_security_group_association" "nsg-assoc" {
  for_each                  = try(var.subnets, {})
  subnet_id                 = azurerm_subnet.snet[each.key].id
  network_security_group_id = azurerm_network_security_group.nsg[each.key].id
}


# route_table to dirvert traffic through Firewall
resource "azurerm_route_table" "rtout" {
  name                = "route-network-outbound"
  resource_group_name = local.rg_name
  location            = data.azurerm_virtual_network.vnet.location
  tags                = merge({ "ResourceName" = "route-network-outbound" }, var.tags, )
}

resource "azurerm_subnet_route_table_association" "rtassoc" {
  for_each = try(var.subnets, {})
  depends_on = [
    azurerm_route_table.rtout,
  ]
  subnet_id      = azurerm_subnet.snet[each.key].id
  route_table_id = azurerm_route_table.rtout.id
}

resource "azurerm_route" "rt" {
  depends_on = [
    azurerm_firewall.fw,
  ]
  name                   = lower("route-to-firewall-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}")
  resource_group_name    = local.rg_name
  route_table_name       = azurerm_route_table.rtout.name
  address_prefix         = "0.0.0.0/0"
  next_hop_type          = "VirtualAppliance"
  next_hop_in_ip_address = azurerm_firewall.fw.ip_configuration.0.private_ip_address
}

# Private DNS Zone - Default is "true"
resource "azurerm_private_dns_zone" "dz" {
  for_each            = try(var.private_dns_zone_name, {})
  name                = lower(each.key)
  resource_group_name = local.rg_name
  tags                = merge({ "ResourceName" = format("%s", lower(var.private_dns_zone_name)) }, var.tags, )
}

resource "azurerm_private_dns_zone_virtual_network_link" "dzvlink" {
  for_each              = try(var.private_dns_zone_name, {})
  name                  = lower("${each.key}-link")
  resource_group_name   = local.rg_name
  virtual_network_id    = data.azurerm_virtual_network.vnet.id
  private_dns_zone_name = azurerm_private_dns_zone.dz[0].name
  tags                  = merge({ "ResourceName" = format("%s", lower("${var.private_dns_zone_name}-link")) }, var.tags, )
}


# Public IP resources for Azure Firewall
resource "random_string" "str" {
  count   = 20
  length  = 6
  special = false
  upper   = false
  keepers = {
    domain_name_label = count.index
  }
}


resource "azurerm_public_ip_prefix" "pip_prefix" {
  name                = lower("${local.vnet_name}-pip-prefix")
  location            = data.azurerm_virtual_network.vnet.location
  resource_group_name = local.rg_name
  prefix_length       = 28
  tags                = merge({ "ResourceName" = lower("${local.vnet_name}-pip-prefix") }, var.tags, )
}

output "prefix" {
  value = azurerm_public_ip_prefix.pip_prefix
}

resource "azurerm_public_ip" "fw-pip" {
  count               = 12
  name                = lower("pip-az-${var.region}-hub01-${data.azurerm_virtual_network.vnet.location}${count.index}")
  location            = data.azurerm_virtual_network.vnet.location
  resource_group_name = local.rg_name
  allocation_method   = "Static"
  sku                 = "Standard"
  public_ip_prefix_id = azurerm_public_ip_prefix.pip_prefix.id
  domain_name_label   = format("%s%s", lower(replace("azfw-${var.region}-hub01", "/[[:^alnum:]]/", "")), random_string.str[count.index].result)
  tags                = merge({ "ResourceName" = lower("pip-az-${var.region}-hub01-${data.azurerm_virtual_network.vnet.location}${count.index}") }, var.tags, )
}
# Azure Firewall 
resource "azurerm_firewall" "fw" {
  name                = lower("fw-${local.vnet_name}")
  location            = data.azurerm_virtual_network.vnet.location
  resource_group_name = local.rg_name
  zones               = var.firewall_zones
  tags                = merge({ "ResourceName" = lower("fw-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}") }, var.tags, )
  dynamic "ip_configuration" {
    for_each = local.fw_pip_configs
    iterator = ip
    content {
      name                 = "azfw-${var.region}-hub01-${split("/", ip.key)[8]}"
      subnet_id            = length(regexall("${data.azurerm_virtual_network.vnet.location}0", split("/", ip.key)[8])) > 0 ? "${data.azurerm_virtual_network.vnet.id}/subnets/AzureFirewallSubnet" : null
      public_ip_address_id = ip.key
    }
  }
}



# Azure Firewall Network/Application/NAT Rules 
resource "azurerm_firewall_application_rule_collection" "fw_app" {
  for_each            = local.fw_application_rules
  name                = lower(format("fw-app-rule-%s-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}", each.key))
  azure_firewall_name = azurerm_firewall.fw.name
  resource_group_name = local.rg_name
  priority            = 100 * (each.value.idx + 1)
  action              = each.value.rule.action

  rule {
    name             = each.key
    source_addresses = each.value.rule.source_addresses
    target_fqdns     = each.value.rule.target_fqdns

    protocol {
      type = each.value.rule.protocol.type
      port = each.value.rule.protocol.port
    }
  }
}

resource "azurerm_firewall_network_rule_collection" "fw" {
  for_each            = local.fw_network_rules
  name                = lower(format("fw-net-rule-%s-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}", each.key))
  azure_firewall_name = azurerm_firewall.fw.name
  resource_group_name = local.rg_name
  priority            = 100 * (each.value.idx + 1)
  action              = each.value.rule.action

  rule {
    name                  = each.key
    source_addresses      = each.value.rule.source_addresses
    destination_ports     = each.value.rule.destination_ports
    destination_addresses = [for dest in each.value.rule.destination_addresses : contains(var.public_ip_names, dest) ? azurerm_public_ip.fw-pip[dest].ip_address : dest]
    protocols             = each.value.rule.protocols
  }
}

resource "azurerm_firewall_nat_rule_collection" "fw" {
  for_each            = local.fw_nat_rules
  name                = lower(format("fw-nat-rule-%s-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}", each.key))
  azure_firewall_name = azurerm_firewall.fw.name
  resource_group_name = local.rg_name
  priority            = 100 * (each.value.idx + 1)
  action              = each.value.rule.action

  rule {
    name                  = each.key
    source_addresses      = each.value.rule.source_addresses
    destination_ports     = each.value.rule.destination_ports
    destination_addresses = [for dest in each.value.rule.destination_addresses : contains(var.public_ip_names, dest) ? azurerm_public_ip.fw-pip[dest].ip_address : dest]
    protocols             = each.value.rule.protocols
    translated_address    = each.value.rule.translated_address
    translated_port       = each.value.rule.translated_port
  }
}


// # Storage Account for Logs Archive
// resource "azurerm_storage_account" "storeacc" {
//   name                      = format("stdiaglogs%s", lower(replace(local.vnet_name, "/[[:^alnum:]]/", "")))
//   resource_group_name       = local.rg_name
//   location                  = data.azurerm_virtual_network.vnet.location
//   account_kind              = "StorageV2"
//   account_tier              = "Standard"
//   account_replication_type  = "GRS"
//   enable_https_traffic_only = true
//   tags                      = merge({ "ResourceName" = format("stdiaglogs%s", lower(replace(local.vnet_name, "/[[:^alnum:]]/", ""))) }, var.tags, )
// }


# Log analytics workspace  for Logs analysis
resource "random_string" "main" {
  length  = 8
  special = false
  keepers = {
    name = local.vnet_name
  }
}

// resource "azurerm_log_analytics_workspace" "logws" {
//   name                = lower("logaws-${random_string.main.result}-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}")
//   resource_group_name = local.rg_name
//   location            = data.azurerm_virtual_network.vnet.location
//   sku                 = var.log_analytics_workspace_sku
//   retention_in_days   = var.log_analytics_logs_retention_in_days
//   tags                = merge({ "ResourceName" = lower("logaws-${random_string.main.result}-${local.vnet_name}-${data.azurerm_virtual_network.vnet.location}") }, var.tags, )
// }


// # Network flow logs for subnet and NSG
// resource "azurerm_network_watcher_flow_log" "nwflog" {
//   for_each                  = var.subnets
//   network_watcher_name      = azurerm_network_watcher.nwatcher[0].name
//   resource_group_name       = azurerm_resource_group.nwatcher[0].name # Must provide Netwatcher resource Group
//   network_security_group_id = azurerm_network_security_group.nsg[each.key].id
//   storage_account_id        = azurerm_storage_account.storeacc.id
//   enabled                   = true
//   version                   = 2
//   retention_policy {
//     enabled = true
//     days    = 0
//   }

//   traffic_analytics {
//     enabled               = true
//     workspace_id          = azurerm_log_analytics_workspace.logws.workspace_id
//     workspace_region      = data.azurerm_virtual_network.vnet.location
//     workspace_resource_id = azurerm_log_analytics_workspace.logws.id
//     interval_in_minutes   = 10
//   }
// }



resource "random_string" "strvgw" {
  length  = 6
  special = false
  upper   = false
  keepers = {
    domain_name_label = var.vpn_gateway_name == "" ? local.vgw_name : var.vpn_gateway_name
  }
}


# Public IP for Virtual Network Gateway
resource "azurerm_public_ip" "pip_gw" {
  name                = var.vpn_gateway_name == "" ? lower("${local.vgw_name}-pip") : lower("${var.vpn_gateway_name}-${data.azurerm_resource_group.rgrp.location}-gw-pip")
  location            = data.azurerm_resource_group.rgrp.location
  resource_group_name = local.rg_name
  allocation_method   = var.public_ip_allocation_method
  sku                 = var.public_ip_sku
  domain_name_label   = format("gw%s%s", lower(replace(var.vpn_gateway_name, "/[[:^alnum:]]/", "")), random_string.strvgw.result)
  tags                = merge({ "ResourceName" = lower("${var.vpn_gateway_name == "" ? lower("${local.vgw_name}-pip") : lower("${var.vpn_gateway_name}-${data.azurerm_resource_group.rgrp.location}-gw-pip")}") }, var.tags, )
}


# Virtual Network Gateway Option For S2S
resource "azurerm_virtual_network_gateway" "vpn" {
  count = var.connections["vnet_gateway_struct"].express_route == false ? 1 : 0
  name                = "${var.connections["vnet_gateway_struct"].name}-vpn"
  resource_group_name = local.rg_name
  location            = data.azurerm_resource_group.rgrp.location
  type                = var.gateway_type
  vpn_type            = var.connections["vnet_gateway_struct"].vpn_type
  sku                 = var.vpn_gw_sku
  active_active       = var.vpn_gw_sku != "Basic" ? var.enable_active_active : false
  enable_bgp          = var.vpn_gw_sku != "Basic" ? var.enable_bgp : false
  generation          = var.vpn_gw_generation


  dynamic "bgp_settings" {
    for_each = var.enable_bgp ? [true] : []
    content {
      asn = var.bgp_asn_number
      #      peering_address = var.bgp_peering_address
      peer_weight = var.bgp_peer_weight
    }
  }

  ip_configuration {
    name                          = "vnetGatewayConfig"
    public_ip_address_id          = azurerm_public_ip.pip_gw.id
    private_ip_address_allocation = "Dynamic"
    subnet_id                     = azurerm_subnet.gw-snet.id
  }

  dynamic "ip_configuration" {
    for_each = var.enable_active_active ? [true] : []
    content {
      name                          = "vnetGatewayAAConfig"
      public_ip_address_id          = azurerm_public_ip.pip_gw.id
      private_ip_address_allocation = "Dynamic"
      subnet_id                     = azurerm_subnet.gw-snet.id
    }
  }

  dynamic "vpn_client_configuration" {
    for_each = var.vpn_client_configuration != null ? [var.vpn_client_configuration] : []
    iterator = vpn
    content {
      address_space = [vpn.value.address_space]
      root_certificate {
        name             = "point-to-site-root-certifciate"
        public_cert_data = vpn.value.certificate
      }
      vpn_client_protocols = vpn.value.vpn_client_protocols
    }
  }
  tags = merge({ "ResourceName" = "${var.vpn_gateway_name == "" ? local.vgw_name : var.vpn_gateway_name}" }, var.tags, )
}

# Virtual Network Gateway Option For Express Route
resource "azurerm_virtual_network_gateway" "er" {
  count               = var.connections["vnet_gateway_struct"].express_route ? 1 : 0
  name                = "${var.connections["vnet_gateway_struct"].name}-er"
  resource_group_name = local.rg_name
  location            = data.azurerm_resource_group.rgrp.location
  type                = var.gateway_type
  vpn_type            = var.connections["vnet_gateway_data"].vpn_type
  sku                 = var.expressroute_sku
  active_active       = var.vpn_gw_sku != "Basic" ? var.enable_active_active : false
  enable_bgp          = var.vpn_gw_sku != "Basic" ? var.enable_bgp : false
  generation          = var.vpn_gw_generation


  dynamic "bgp_settings" {
    for_each = var.enable_bgp ? [true] : []
    content {
      asn = var.bgp_asn_number
      #      peering_address = var.bgp_peering_address
      peer_weight = var.bgp_peer_weight
    }
  }

  ip_configuration {
    name                          = "vnetGatewayConfig"
    public_ip_address_id          = azurerm_public_ip.pip_gw.id
    private_ip_address_allocation = "Dynamic"
    subnet_id                     = azurerm_subnet.gw-snet.id
  }

  dynamic "ip_configuration" {
    for_each = var.enable_active_active ? [true] : []
    content {
      name                          = "vnetGatewayAAConfig"
      public_ip_address_id          = azurerm_public_ip.pip_gw.id
      private_ip_address_allocation = "Dynamic"
      subnet_id                     = azurerm_subnet.gw-snet.id
    }
  }

  dynamic "vpn_client_configuration" {
    for_each = var.vpn_client_configuration != null ? [var.vpn_client_configuration] : []
    iterator = vpn
    content {
      address_space = [vpn.value.address_space]
      root_certificate {
        name             = "point-to-site-root-certifciate"
        public_cert_data = vpn.value.certificate
      }
      vpn_client_protocols = vpn.value.vpn_client_protocols
    }
  }
  #  tags = merge({ "ResourceName" = "${var.vpn_gateway_name == "" ? local.vgw_name : var.vpn_gateway_name}" }, var.tags, )
}

# Local Network Gateway VPN
resource "azurerm_local_network_gateway" "local_gw" {
  for_each = var.connections["connect_struct"]
  name                = each.value["local_gateway_data"].name
  resource_group_name = local.rg_name
  location            = data.azurerm_resource_group.rgrp.location
  gateway_address     = each.value["local_gateway_data"].local_gateway_address
  address_space       = each.value["local_gateway_data"].local_address_space

  dynamic "bgp_settings" {
    for_each = var.local_bgp_settings != null ? var.local_bgp_settings : null
    content {
      asn                 = bgp_settings.value["asn_number"]
      bgp_peering_address = bgp_settings.value["peering_address"]
      peer_weight         = bgp_settings.value["peer_weight"]
    }
  }
  #  tags = merge({ "ResourceName" = "localgw-${var.local_networks[count.index].local_gw_name}" }, var.tags, )
}


# Virtual Network Gateway Connection
resource "azurerm_virtual_network_gateway_connection" "az-hub-onprem" {
  for_each                        = var.connections["connect_struct"]
  name                            = replace(each.value["local_gateway_data"].name, "localgtw-", "conn-")
  resource_group_name             = local.rg_name
  location                        = data.azurerm_resource_group.rgrp.location
  type                            = each.value["connection_type"]
  virtual_network_gateway_id      = each.value["connection_type"] == "ExpressRoute" ? azurerm_virtual_network_gateway.er[0].id : azurerm_virtual_network_gateway.vpn[0].id
  local_network_gateway_id        = each.value["connection_type"] == "ExpressRoute" ? azurerm_local_network_gateway.local_gw[each.key].id : azurerm_local_network_gateway.local_gw[each.key].id
  express_route_circuit_id        = each.value["connection_type"] == "ExpressRoute" ? var.express_route_circuit_id : null
  peer_virtual_network_gateway_id = each.value["connection_type"] == "Vnet2Vnet" ? var.peer_virtual_network_gateway_id : null
  shared_key                      = each.value["shared_key"]
  connection_protocol = var.gateway_connection_type == "IPSec" && var.vpn_gw_sku == ["VpnGw1", "VpnGw2", "VpnGw3", "VpnGw1AZ", "VpnGw2AZ", "VpnGw3AZ"] ? var.gateway_connection_protocol : null

  ipsec_policy {
    dh_group         = var.connections["ipsec_policy"].dh_group
    ike_encryption   = var.connections["ipsec_policy"].ike_encryption
    ike_integrity    = var.connections["ipsec_policy"].ike_integrity
    ipsec_encryption = var.connections["ipsec_policy"].ipsec_encryption
    ipsec_integrity  = var.connections["ipsec_policy"].ipsec_integrity
    pfs_group        = var.connections["ipsec_policy"].pfs_group
    sa_datasize      = var.connections["ipsec_policy"].sa_datasize
    sa_lifetime      = var.connections["ipsec_policy"].sa_lifetime
  }
  // tags = merge({ "ResourceName" = var.gateway_connection_type == "ExpressRoute" ? "localgw-expressroute-connection" : "localgw-connection-${var.local_networks[count.index].local_gw_name}" }, var.tags, )
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
Error: Plugin did not respond

  with module.pd[0].azurerm_public_ip_prefix.pip_prefix,
  on modules/hub_resources/main.tf line 177, in resource "azurerm_public_ip_prefix" "pip_prefix":
 177: resource "azurerm_public_ip_prefix" "pip_prefix" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Error: Plugin did not respond

  with module.pd[0].azurerm_public_ip.pip_gw,
  on modules/hub_resources/main.tf line 346, in resource "azurerm_public_ip" "pip_gw":
 346: resource "azurerm_public_ip" "pip_gw" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Error: Plugin did not respond

  with module.pd[0].azurerm_local_network_gateway.local_gw["telefonica"],

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Stack trace from the terraform-provider-azurerm_v2.96.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

```

### Expected Behaviour

* Terraform should have deployed said resources

### Actual Behaviour

* While running on a previously successful pipeline in Azure DevOps the Terraform plugin for 2.96 has started crashing every time starting at 1 PM PST

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

* Could this be an issue with hosted agents inside of Azure DevOps?

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #14698

--- a/internal/services/network/migration/network_watcher_flow_log.go
+++ b/internal/services/network/migration/network_watcher_flow_log.go
@@ -143,7 +143,23 @@ func (NetworkWatcherFlowLogV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
 			return rawState, err
 		}
 
-		id := parse.NewFlowLogID(watcherId.SubscriptionId, watcherId.ResourceGroup, watcherId.Name, rawState["name"].(string))
+		var name string
+		rawName, ok := rawState["name"]
+		if ok {
+			name = rawName.(string)
+		} else {
+			// The `name` is introduced as an attribute since 0e528be. If users have provisioned this resource prior to that commit, and didn't run a `refresh` for the flow log. Then the state won't have `name` included.
+			// In this case, we will use the Portal way to construct the flow log name.
+			nsgId, err := parse.NetworkSecurityGroupID(parts[1])
+			if err != nil {
+				return rawState, err
+			}
+			name = fmt.Sprintf("Microsoft.Network%s%s", watcherId.ResourceGroup, nsgId.Name)
+			if len(name) > 80 {
+				name = name[:80]
+			}
+		}
+		id := parse.NewFlowLogID(watcherId.SubscriptionId, watcherId.ResourceGroup, watcherId.Name, name)
 		newId := id.ID()
 		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
 		rawState["id"] = newId

